turbo/snapshotsync/freezeblocks/bor_snapshots.go

```diff
@@ -75,12 +75,20 @@ func (br *BlockRetire) retireBorBlocks(ctx context.Context, minBlockNum uint64,
 				return false, nil
 			}
 
-			logger.Log(lvl, "[bor snapshots] Retire Bor Blocks", "type", snaptype,
+			logger.Log(lvl, "[bor snapshots] Retire Bor Blocks", "type", snap,
 				"range", fmt.Sprintf("%s-%s", common.PrettyCounter(blockFrom), common.PrettyCounter(blockTo)))
 
-			for i := blockFrom; i < blockTo; i = chooseSegmentEnd(i, blockTo, snaptype.Enum(), chainConfig) {
-				end := chooseSegmentEnd(i, blockTo, snaptype.Enum(), chainConfig)
-				if _, err := snaptype.ExtractRange(ctx, snaptype.FileInfo(snapshots.Dir(), i, end), nil, db, chainConfig, tmpDir, workers, lvl, logger); err != nil {
+			var firstKeyGetter snaptype.FirstKeyGetter
+
+			if snap.Enum() == borsnaptype.BorEvents.Enum() {
+				firstKeyGetter = func(ctx context.Context) uint64 {
+					return max(1, blockReader.LastFrozenEventId())
```

Comment:
double checking if this is correct because it looks like an off by one error

should not it be `max(1, blockReader.LastFrozenEventId()+1)`?

blockReader.LastFrozenEventId() returns the last event id which is already frozen. So we need to start at the one after it.

Reply:
In the snapshots format the `blockReader.LastFrozenEventId()` is still the first event ID of the next block, so this is correct.